c-dependencies/js-compute-runtime/builtins/backend.h

```diff
@@ -1,8 +1,273 @@
 #ifndef JS_COMPUTE_RUNTIME_BACKEND_H
 #define JS_COMPUTE_RUNTIME_BACKEND_H
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
 
 #include "builtin.h"
 
+enum class Authentication {
+  RSA,
+};
+
+enum class KeyExchange {
+  EECDH,
+  RSA,
+};
+
+enum class Encryption {
+  AES128,
+  AES128GCM,
+  AES256,
+  AES256GCM,
+  CHACHA20POLY1305,
+  TRIPLE_DES,
+};
+
+enum class EncryptionLevel {
+  MEDIUM,
+  HIGH,
+};
+
+enum class MessageDigest {
+  SHA1,
+  SHA256,
+  SHA384,
+  AEAD,
+};
+
+enum class Protocol {
+  SSLv3,
+  TLSv1,
+  TLSv1_2,
+};
+
+class Cipher {
+  std::string id;
+  std::string openSSLAlias;
+  KeyExchange kx;
+  Authentication au;
+  Encryption enc;
+  MessageDigest mac;
+  Protocol protocol;
+  EncryptionLevel level;
+  int strength_bits;
+
+public:
+  Cipher(std::string openSSLAlias, KeyExchange kx, Authentication au, Encryption enc,
+         MessageDigest mac, Protocol protocol, EncryptionLevel level, int strength_bits)
+      : openSSLAlias(openSSLAlias), kx(kx), au(au), enc(enc), mac(mac), protocol(protocol),
+        level(level), strength_bits(strength_bits) {}
+
+  std::string getOpenSSLAlias() { return openSSLAlias; }
+
+  KeyExchange getKx() { return kx; }
+
+  Authentication getAu() { return au; }
+
+  Encryption getEnc() { return enc; }
+
+  MessageDigest getMac() { return mac; }
+
+  Protocol getProtocol() { return protocol; }
+
+  EncryptionLevel getLevel() { return level; }
+
+  int getStrength_bits() { return strength_bits; }
+
+  // Overload the == operator
+  const bool operator==(const Cipher &obj) const {
+    return openSSLAlias == obj.openSSLAlias && kx == obj.kx && au == obj.au && enc == obj.enc &&
+           mac == obj.mac && protocol == obj.protocol && level == obj.level &&
+           strength_bits == obj.strength_bits;
+  }
+};
+
+/**
+ * Class in charge with parsing openSSL expressions to define a list of ciphers.
+ */
+class OpenSSLCipherConfigurationParser {
+public:
+  // inline static std::map<std::string, std::vector<Cipher>> aliases;
+  inline static bool initialized = false;
```

Comment:
If you need to mutate this class variable, it shouldn't be marked `inline`. I would instead initialize it at the top level in backend.cpp as `bool OpenSSLCipherConfigurationParser::initialized = false;`. Another option would be to make all the static methods instance methods instead, and make a single top-level value of `OpenSSLCipherConfigurationParser` defined in an anonymous namespace in backend.cpp. That way you wouldn't need to worry about where to initialize class variables, as any state to be mutated would be an instance variable.
```suggestion
  static bool initialized;
```

Reply:
I've refactored to using `std::once_flag` and `std::call_once` instead, which seems really nice